### PR TITLE
Problem: (fix #1493) no way to generate test vectors in dev-utils

### LINKED ISSUE #1493: Problem: no way to generate test vectors in dev-utils
the complementary tool part for https://github.com/crypto-com/chain-docs/issues/68

### PR DESCRIPTION
**Pull request by linfeng-crypto**
the result is:

```
{
  "withdraw_unbounded_vector": {
    "transfer_address": "cro1duepg06jrlgpdphxlnfrt3lxu95mdpl5g03vxn89yzhvdl74p6ms4ursr6",
    "staking_address": "cro13xh025aqd2cvx9e7080pecjp48knhxfv40entz",
    "witness": "000149553434c87cdb2fa8fbfb251d83143401077f3d09b9ed770f43493ce35abf7920332c095b599235b5088e8188de3f45b296c30f55405c0ddff6d71bf3d28ce1",
    "payload": "00000000000000000000000000000000000000003d040d0f0f0f0f0f0f0f0f1f0f603d1b305d10df1989e9f3dd3ac8e9ee66b988fb4cedcc43ea2fa1c9f0da01b80feefa0a0f0f0f0f0e0f0f0f0f0f0f0f0f0f603d1b305d10df1989e9f3dd3ac8e9ee66b988fb4cedcc43ea2fa1c9f0da01b80fcde4040f0f0f0f0e0f0f0f0f0f0f0f0f0f603d1b305d10df1989e9f3dd3ac8e9ee66b988fb4cedcc43ea2fa1c9f0da01b80feefa0a0f0f0f0f0e0f0f0f0f0f0f0f0f0f603d1b305d10df1989e9f3dd3ac8e9ee66b988fb4cedcc43ea2fa1c9f0da01b87d751f7d0b2cc8850e0f0f0f0f0f0f0f0f0f0f0b0db68570b7c30f7f476d546b49a246aebca82dd08313a67ab77e6f0d311bde9f980f0f0f0f0f0f0f0f0f000000000000000000000000000000004408094749c7b5545870ced0aa03898d8f81eaef54433dbc0f24dab1b9914cfc",
    "tx": "00020400000149553434c87cdb2fa8fbfb251d83143401077f3d09b9ed770f43493ce35abf7920332c095b599235b5088e8188de3f45b296c30f55405c0ddff6d71bf3d28ce100000000000000000000000000000000000000003d040d0f0f0f0f0f0f0f0f1f0f603d1b305d10df1989e9f3dd3ac8e9ee66b988fb4cedcc43ea2fa1c9f0da01b80feefa0a0f0f0f0f0e0f0f0f0f0f0f0f0f0f603d1b305d10df1989e9f3dd3ac8e9ee66b988fb4cedcc43ea2fa1c9f0da01b80fcde4040f0f0f0f0e0f0f0f0f0f0f0f0f0f603d1b305d10df1989e9f3dd3ac8e9ee66b988fb4cedcc43ea2fa1c9f0da01b80feefa0a0f0f0f0f0e0f0f0f0f0f0f0f0f0f603d1b305d10df1989e9f3dd3ac8e9ee66b988fb4cedcc43ea2fa1c9f0da01b87d751f7d0b2cc8850e0f0f0f0f0f0f0f0f0f0f0b0db68570b7c30f7f476d546b49a246aebca82dd08313a67ab77e6f0d311bde9f980f0f0f0f0f0f0f0f0f000000000000000000000000000000004408094749c7b5545870ced0aa03898d8f81eaef54433dbc0f24dab1b9914cfc"
  },
  "transfer_vector": {
    "transfer_address": "dcro1duepg06jrlgpdphxlnfrt3lxu95mdpl5g03vxn89yzhvdl74p6msq4uly6",
    "witness": "0400aa27c7dd8867e11b6726e62ddc6cf3479e8f8acd44a6de08d4b42320f0c23b7550cc28793a806e9394b227c5866c0e752cfe8c70e38de4f0765670f2f23527a200b98a7fb8cc007048625b6446ad49a1b3a722df8c1ca975b87160023e14d19097",
    "payload": "000000000000000000000000000000000000000035030f0b4b07064846c8ba5b577fc1dfa50c8682808ee5e05b4c32b3002bd5beb69e43f30f0f0b0f603d1b305d10df1989e9f3dd3ac8e9ee66b988fb4cedcc43ea2fa1c9f0da01b8dbd0fa0a0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0b0fa528c8d28768ee146829e922d363fc48918085c24ba9d107dbbb2c2fffcd347a5fc32776358f619c9bbd28ca8963017a23f1837fec82ebff79597ffdfd3a28ad0fb68570b7c30f7f476d546b49a246aebca82dd08313a67ab77e6f0d311bde9f98000000000000000000000000000000006e0785c2d6f032dcb6422012059b57c2a09c4adad269f8eee58f196844aecff4",
    "tx": "0000044408094749c7b5545870ced0aa03898d8f81eaef54433dbc0f24dab1b9914cfc00000100000000000000000000000000000000000000000035030f0b4b07064846c8ba5b577fc1dfa50c8682808ee5e05b4c32b3002bd5beb69e43f30f0f0b0f603d1b305d10df1989e9f3dd3ac8e9ee66b988fb4cedcc43ea2fa1c9f0da01b8dbd0fa0a0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0b0fa528c8d28768ee146829e922d363fc48918085c24ba9d107dbbb2c2fffcd347a5fc32776358f619c9bbd28ca8963017a23f1837fec82ebff79597ffdfd3a28ad0fb68570b7c30f7f476d546b49a246aebca82dd08313a67ab77e6f0d311bde9f98000000000000000000000000000000006e0785c2d6f032dcb6422012059b57c2a09c4adad269f8eee58f196844aecff4"
  },
  "deposit_stake_vector": {
    "staking_address": "cro13xh025aqd2cvx9e7080pecjp48knhxfv40entz",
    "witness": "040016b4508ed0cfaa7d8ddb6a66acf74f161e8ba1e9ca2ee14bf73307a8fdd35d8af8167f9b39b077453e1dc41a53fb594abbd65f33ed1990e33566acb366b9b63c00b98a7fb8cc007048625b6446ad49a1b3a722df8c1ca975b87160023e14d19097",
    "payload": "0000000000000000000000000000000000000000d1010e0b0f19bb5f81dfc0a57282d46569a3f840191184aee6c521ee44f83c08a7f2dc5285f719709436bf784a3112cb155cf45645b4d9503ce2169fec3a69a3bc69b6b9330fb68570b7c30f7f476d546b49a246aebca82dd08313a67ab77e6f0d311bde9f9800000000000000000000000000000000a448edc74661a857ea053eee16a79863339ea1cb90359fe23dedcc1d7da626f6",
    "tx": "0001044408094749c7b5545870ced0aa03898d8f81eaef54433dbc0f24dab1b9914cfc01000089aef553a06ab0c3173e79de1ce241a9ed3b992c000000000000000000000000000000000000000000000000000000000000d1010e0b0f19bb5f81dfc0a57282d46569a3f840191184aee6c521ee44f83c08a7f2dc5285f719709436bf784a3112cb155cf45645b4d9503ce2169fec3a69a3bc69b6b9330fb68570b7c30f7f476d546b49a246aebca82dd08313a67ab77e6f0d311bde9f9800000000000000000000000000000000a448edc74661a857ea053eee16a79863339ea1cb90359fe23dedcc1d7da626f6"
  },
  "nodejoin_vector": {
    "staking_address": "cro13xh025aqd2cvx9e7080pecjp48knhxfv40entz",
    "witness": "00017a909fd00b43678183544dd5d67a82a472f072f1aa41e63b845b195be13f4cd32cde5882411643dd0006f151964403f401f5b1f43622bc8f63e5f52d56b9f499",
    "tx": "010201000000000000000089aef553a06ab0c3173e79de1ce241a9ed3b992c00000000000000000000001c6e6f2d6e616d6500000202020202020202020202020202020202020202020202020202020202020202144649584d4500017a909fd00b43678183544dd5d67a82a472f072f1aa41e63b845b195be13f4cd32cde5882411643dd0006f151964403f401f5b1f43622bc8f63e5f52d56b9f499"
  },
  "unbounded_stake_vector": {
    "staking_address": "cro13xh025aqd2cvx9e7080pecjp48knhxfv40entz",
    "witness": "0000c34cc5776366939e7c6fa6b34f7f054e787f25bebbf1da431bc2eb1b0de8465b2901677dddf463aa355692fff9e330a5c236e5ed03d0cca92a51213cf693714d",
    "tx": "01000089aef553a06ab0c3173e79de1ce241a9ed3b992c02000000000000000100000000000000000000000000000000000000c34cc5776366939e7c6fa6b34f7f054e787f25bebbf1da431bc2eb1b0de8465b2901677dddf463aa355692fff9e330a5c236e5ed03d0cca92a51213cf693714d"
  }
}
```

is the result struct Ok? or should I add some other information like key-pair.

### CODE REVIEW

File: dev-utils/src/commands/test_vector_command.rs [hunk 1]
```diff
@@ -0,0 +1,595 @@
...
+use chain_core::tx::data::{Tx, TxId};
+use chain_core::tx::fee::{LinearFee, Milli};
+use chain_core::tx::witness::tree::RawXOnlyPubkey;
+use chain_core::tx::witness::{EcdsaSignature, TxInWitness, TxWitness};
+use chain_core::tx::TransactionId;
+use chain_core::tx::{PlainTxAux, TxAux, TxEnclaveAux, TxPublicAux};
+use chain_storage::buffer::Get;
+use chain_storage::{LookupItem, Storage, NUM_COLUMNS};
+use client_common::Result;
+use mock_utils::encrypt;
```
Comment (yihuang) [inline dev-utils/src/commands/test_vector_command.rs:47]: The `mock_utils` is only used for development conveniences on non-sgx platforms, the obfustication result is different from the real platform.

File: dev-utils/src/dev_utils.rs [hunk 2]
```diff
@@ -35,6 +36,21 @@ pub enum DevUtils {
     /// Used for stopping
     #[structopt(name = "stop", about = "stop all chain components")]
     Stop,
+
+    /// Create Test Vector
+    #[structopt(name = "test-vectors", about = "create test vector")]
```
Comment (leejw51crypto) [inline dev-utils/src/dev_utils.rs:41]: how about just `test`?

File: dev-utils/src/commands/test_vector_command.rs (hunk 1 above)
Comment (linfeng-crypto) [reply dev-utils/src/commands/test_vector_command.rs:47]: yes, I'll try to use the real one.

File: dev-utils/src/dev_utils.rs (hunk 2 above)
Comment (tomtau) [reply dev-utils/src/dev_utils.rs:41]: test vectors is more precise

File: dev-utils/Cargo.toml [hunk 3]
```diff
@@ -6,20 +6,29 @@ description = "Basic CLI for development purposes (e.g. generation of genesis.js
 edition = "2018"
 
 [dependencies]
+abci = "0.7"
+bit-vec = { version = "0.6.1", features = ["serde_no_std"] }
```
Comment (tomtau) [inline dev-utils/Cargo.toml:10]: does it need all these extra dependencies?

Comment (tomtau) [reply dev-utils/Cargo.toml:10]: and chain-storage, chrono, dirs...

**Pull request merged**